course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/PageManagementActor.java

```diff
@@ -843,4 +846,144 @@ private void validateOrg(String orgId) {
+                          List<Map<String, Object>> sectionList = getUserProfileData(Lists.newArrayList(sections), userProfile);
+                          Map<String, Object> result = new HashMap<>();
+                          result.put(JsonKey.NAME, finalPageMap.get(JsonKey.NAME));
+                          result.put(JsonKey.ID, finalPageMap.get(JsonKey.ID));
+                          result.put(JsonKey.SECTIONS, sectionList);
+                          result.put("ignoredSections", ignoredSections);
+                          Response response = new Response();
+                          response.put(JsonKey.RESPONSE, result);
+                          ProjectLogger.log(
+                                  "PageManagementActor:getPageData:apply: Response before caching it = "
+                                          + response,
+                                  LoggerEnum.INFO);
+                          return response;
+                        }
+                      },
+                      getContext().dispatcher());
+      Patterns.pipe(response, getContext().dispatcher()).to(sender());
+    }
+    catch (Exception e) {
+      ProjectLogger.log(
+              "PageManagementActor:getPageData: Exception occurred with error message = "
+                      + e.getMessage(),
+              LoggerEnum.ERROR);
+      ProjectLogger.log(
+              "PageManagementActor:getPageData: Exception occurred with error message = "
+                      + e.getMessage(),
+              e);
+    }
+  }
+
+  private List<Map<String, Object>> getUserProfileData(List<Map<String, Object>> sectionList, Map<String, Object> userProfile) {
+    if(MapUtils.isEmpty(userProfile)) {
+      return sectionList;
+    } else {
+      // filter all sections containing collectionCount > 1
+      List<Map<String, Object>> filteredSections = sectionList.stream().filter(section -> (Integer) section.getOrDefault("collectionsCount", 0) > 0).collect(Collectors.toList());
+      Map<String, Object> filteredUserProfile = userProfilePropList.stream().collect(Collectors.toMap(key -> key, key -> userProfile.get(key)));
+      filteredUserProfile.values().removeIf(Objects::isNull);
+      // filter collections containing shallow copy && having board given in the userProfile,
+      if(CollectionUtils.isNotEmpty(filteredSections) && MapUtils.isNotEmpty(filteredUserProfile)){
+        for(Map<String, Object> section: filteredSections) {
+          List<Map<String, Object>> collections = (List<Map<String, Object>>) section.get("collections");
+          List<Map<String, Object>> shallowCopied = collections.stream().filter(content -> ((String)content.getOrDefault("originData", "")).contains("shallow")).collect(Collectors.toList());
+          List<Map<String, Object>> originCollections = collections.stream().filter(content -> !content.containsKey("originData")).collect(Collectors.toList());
```

Comment:
We are not handling deep copied collections. They also should be part of `originCollections`.